ci/scripts/cpp_test.sh

```diff
@@ -86,6 +86,7 @@ ctest \
     --label-regex unittest \
     --output-on-failure \
     --parallel ${n_jobs} \
+    --repeat until-pass:3 \
```

Comment:
Why is this necessary? To prevent flakeyness due to network timeout? Are we running tests against remote buckets or just azurite?